node: don't set write-only NodeAddressingElement.AddressType property

tklauser opened the pull request:
This field is only ever set but never read (i.e. inside the CNI). We
need to keep the property as part of the API because of backwards
compatibility, but it's fine to no longer set it.
    
This allows to drop `pkg/node.getNodeIP` and fold it into
`pkg/node.GetNodeIP`.

rolinh commented on api/v1/openapi.yaml:
```diff
@@ -2327,9 +2327,6 @@ definitions:
         type: string
-      address-type:
-        description: Node address type, one of HostName, ExternalIP or InternalIP
-        type: string
```
In [the doc](https://docs.cilium.io/en/latest/api/#compatibility-guarantees), we make the following statement:
> Cilium API is stable as of version 1.0, backward compatibility will be upheld for whole lifecycle of Cilium 1.x.

My understanding is that this change breaks this promise. Could we somehow keep this but make this a no-op and mark deprecated?

tklauser replied:
Right, good point. I seem to remember we had ro revert other changes like this due to failures to unmarshal older versions of certain definitions.

I guess we should keep the field in the API and just no longer set it (i.e. keep this PR's changes to `pkg/node`, but leave the API untouched).

tklauser committed:
node: don't set write-only NodeAddressingElement.AddressType property

This field is only ever set but never read (i.e. inside the CNI). We
need to keep the property as part of the API because of backwards
compatibility, but it's fine to no longer set it.

This allows to drop pkg/node.getNodeIP and fold it into
pkg/node.GetNodeIP.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
